lib/Service/OpenAiAPIService.php

```diff
@@ -121,22 +249,39 @@ public function createCompletion(?string $userId, string $prompt, int $n, string
 	 * @param bool $storePrompt
 	 * @return array|string[]
 	 * @throws \OCP\DB\Exception
+	 * @throws Exception
 	 */
 	public function createChatCompletion(?string $userId, string $prompt, int $n, string $model, int $maxTokens = 1000,
 										 bool $storePrompt = true): array {
+		if($this->isQuotaExceeded($userId, Application::QUOTA_TYPE_TEXT)){
+			return ['error' => 'Text generation quota exceeded'];
```

Comment:
Aaah

Reply:
Fixed in: https://github.com/nextcloud/integration_openai/commit/6697be3b38618938a3fe9e5ab264d1eb7aeec0fe